panther_utils/include/panther_utils/test/test_utils.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef PANTHER_UTILS_TEST_UTILS_HPP_
+#define PANTHER_UTILS_TEST_UTILS_HPP_
+
+#include <chrono>
+#include <memory>
+
+#include <rclcpp/rclcpp.hpp>
+
+using namespace std::chrono_literals;
+
+namespace panther_utils::test_utils
+{
+
+template <typename NodeT, typename MsgT>
+bool WaitForMsg(
+  const std::shared_ptr<NodeT> node, std::shared_ptr<MsgT> & msg,
```

Comment:
```suggestion
  const std::shared_ptr<NodeT> & node, std::shared_ptr<MsgT> & msg,
```
If it is not, reference copy of the node is created on every invocation of this function